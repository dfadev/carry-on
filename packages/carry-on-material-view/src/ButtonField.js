import React from "react";
import Button from "@material-ui/core/Button";
import withStyles from "@material-ui/core/styles/withStyles";
import { Field } from "carry-on-react-forms";
import getFieldStatus from "./getFieldStatus";
import eventHandler from "./eventHandler";

const ButtonField = withStyles(
  {},
  { name: "CoButtonField" }
)(
  ({
    name,
    label = name,
    value: valueProp,
    disabled: disabledProp,
    onClick,
    ...props
  }) => (
    <Field path={name} type="button">
      {(field, store) => {
        const { disabled, value } = getFieldStatus(
          field,
          store,
          valueProp,
          disabledProp
        );

        return (
          <Button
            {...props}
            disabled={disabled}
            {...field.element}
            value={value}
            onClick={eventHandler(
              onClick,
              props.id || field.element.id,
              value,
              store
            )}
          >
            {label}
          </Button>
        );
      }}
    </Field>
  )
);

export default ButtonField;
