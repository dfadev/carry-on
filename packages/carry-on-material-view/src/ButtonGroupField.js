import React from "react";
import withStyles from "@material-ui/core/styles/withStyles";
import Button from "@material-ui/core/Button";
import ButtonGroup from "@material-ui/core/ButtonGroup";
import Grid from "@material-ui/core/Grid";
import { Field } from "carry-on-react-forms";
import getFieldStatus from "./getFieldStatus";
import eventHandler from "./eventHandler";

const ButtonGroupField = withStyles(
  {},
  { name: "CoButtonGroupField" }
)(
  ({
    name,
    label = null,
    value: valueProp,
    disabled: disabledProp,
    buttons
  }) => (
    <Field path={name}>
      {(field, store) => {
        if (!buttons || buttons.length === 0) return null;

        const { disabled, value } = getFieldStatus(
          field,
          store,
          valueProp,
          disabledProp
        );

        let idIndex = 0;

        const buttonGroup = (
          <ButtonGroup value={value} disabled={disabled}>
            {buttons.map(
              ({
                value: val,
                content,
                onClick,
                // eslint-disable-next-line no-plusplus
                id = `${field.element.id}[${idIndex++}]`,
                ...rest
              }) => (
                <Button
                  key={id}
                  id={id}
                  value={val}
                  onClick={eventHandler(onClick, id, val, store)}
                  {...rest}
                >
                  {content}
                </Button>
              )
            )}
          </ButtonGroup>
        );

        if (label !== undefined && label !== null) {
          return (
            <Grid container spacing={1}>
              <Grid item xs={12}>
                {label}
              </Grid>
              <Grid item>{buttonGroup}</Grid>
            </Grid>
          );
        }

        return buttonGroup;
      }}
    </Field>
  )
);

export default ButtonGroupField;
