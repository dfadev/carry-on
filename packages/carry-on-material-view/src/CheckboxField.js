import React from "react";
import Checkbox from "@mui/material/Checkbox";
import FormControl from "@mui/material/FormControl";
import FormControlLabel from "@mui/material/FormControlLabel";
import FormHelperText from "@mui/material/FormHelperText";
import { withStyles } from "tss-react/mui";
import { Field } from "carry-on-react-forms";
import getFieldStatus from "./getFieldStatus";

function CheckboxField({
  name,
  label = name,
  readOnly,
  required,
  disabled: disabledProp,
  value: valueProp,
  labelPlacement,
  ...props
}) {
  return (
    <Field path={name} type="checkbox" readOnly={readOnly}>
      {(field, store) => {
        const { element } = field;
        const { error, hasError, disabled, value } = getFieldStatus(
          field,
          store,
          valueProp,
          disabledProp
        );

        return (
          <FormControl error={hasError} required={required}>
            <FormControlLabel
              control={
                <Checkbox
                  {...props}
                  inputProps={{ ...element, checked: value }}
                  checked={value}
                />
              }
              label={label}
              disabled={disabled}
              labelPlacement={labelPlacement}
            />
            {hasError && <FormHelperText>{error}</FormHelperText>}
          </FormControl>
        );
      }}
    </Field>
  );
}

const StyledCheckboxField = withStyles(
  CheckboxField,
  {},
  { name: "CoCheckboxField" }
);

export default StyledCheckboxField;
