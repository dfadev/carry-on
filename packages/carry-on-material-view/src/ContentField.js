import React from "react";
import withStyles from "@material-ui/core/styles/withStyles";
import { Field } from "carry-on-react-forms";

const ContentField = withStyles(
  {},
  { name: "CoContentField" }
)(({ name, content }) => (
  <Field path={name}>
    {field =>
      content === undefined && content !== null ? field.element.value : content
    }
  </Field>
));

export default ContentField;
