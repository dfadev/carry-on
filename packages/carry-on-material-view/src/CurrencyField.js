import React from "react";
import InputAdornment from "@material-ui/core/InputAdornment";
import withStyles from "@material-ui/core/styles/withStyles";
import MaskedTextField from "./MaskedTextField";

const numberAccept = /[\d.]+/g;

const parseNumber = string => {
  if (typeof string !== "string") string = string.toString();
  return (string.match(numberAccept) || []).join("");
};

const formatFloatingPointNumber = (value, maxDigits) => {
  const parsed = parseNumber(value.slice(0, 19));
  const [head, tail] = parsed.split(".");
  // Avoid rounding errors at toLocaleString as when user enters 1.239 and maxDigits=2 we
  // must not to convert it to 1.24, it must stay 1.23
  const scaledTail = tail != null ? tail.slice(0, maxDigits) : "";

  const number = Number.parseFloat(`${head}.${scaledTail}`);

  if (Number.isNaN(number)) {
    return "";
  }

  const formatted = number.toLocaleString(undefined, {
    minimumFractionDigits: 0,
    maximumFractionDigits: maxDigits
  });

  if (parsed.includes(".")) {
    const [formattedHead] = formatted.split(".");

    // skip zero at digits position for non fixed floats
    // as at digits 2 for non fixed floats numbers like 1.50 has no sense, just 1.5 allowed
    // but 1.0 has sense as otherwise you will not be able to enter 1.05 for example
    const formattedTail =
      scaledTail !== "" &&
      scaledTail.length < maxDigits &&
      scaledTail[maxDigits] === "0"
        ? scaledTail.slice(0, -1)
        : scaledTail;

    return maxDigits === 0
      ? formattedHead
      : `${formattedHead}.${formattedTail}`;
  }
  return formatted;
};

const formatCurrency = maxDigits => string =>
  formatFloatingPointNumber(string, maxDigits);

const accept = /[\d.$]/g;

const CurrencyField = withStyles(
  {},
  { name: "CoCurrencyField" }
)(({ InputProps, maximumFractionDigits = 2, symbol = "$", ...props }) => (
  <MaskedTextField
    accept={accept}
    format={formatCurrency(maximumFractionDigits)}
    InputProps={{
      startAdornment: (
        <InputAdornment position="start">{symbol}</InputAdornment>
      ),
      ...(InputProps || {})
    }}
    {...(props || {})}
  />
));

export default CurrencyField;
