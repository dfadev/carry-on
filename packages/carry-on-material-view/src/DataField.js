import React from "react";
import { withStyles } from "tss-react/mui";
import { Field } from "carry-on-react-forms";

function DataField({ name, value, transform }) {
  return (
    <Field path={name} readOnly>
      {(field, store) => {
        let fieldValue;
        if (value !== undefined) {
          fieldValue =
            typeof value === "function" ? value(field, store) : value;
        } else {
          fieldValue = field.element.value;
        }

        if (transform) fieldValue = transform(fieldValue);

        return fieldValue;
      }}
    </Field>
  );
}

const StyledDataField = withStyles(DataField, {}, { name: "CoDataField" });

export default StyledDataField;
