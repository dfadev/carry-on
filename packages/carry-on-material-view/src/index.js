export { default as ButtonField } from "./ButtonField";
export { default as CheckboxField } from "./CheckboxField";
export { default as CurrencyField } from "./CurrencyField";
export { default as EINField } from "./EINField";
export { default as IconButtonField } from "./IconButtonField";
export { default as MaskedTextField } from "./MaskedTextField";
export { default as PercentageField } from "./PercentageField";
export { default as PhoneField } from "./PhoneField";
export { default as RegisterMaterialView } from "./RegisterMaterialView";
export { default as ResetButtonField } from "./ResetButtonField";
export { default as SSNField } from "./SSNField";
export { default as SubmitButtonField } from "./SubmitButtonField";
export { default as TextField } from "./TextField";
export { default as ZipCodeField } from "./ZipCodeField";
export { default as getFieldStatus } from "./getFieldStatus";
export {
  default as registerComponents,
  materialViewComponents
} from "./registerComponents";
export { default as SelectField } from "./SelectField";
export { default as RadioField } from "./RadioField";
export { default as SwitchField } from "./SwitchField";
export { default as ToggleButtonGroupField } from "./ToggleButtonGroupField";
export { default as ContentField } from "./ContentField";
export { default as InspectorField } from "./InspectorField";
export { default as DateField } from "./DateField";
export { default as DobField } from "./DobField";
export { default as TimeField } from "./TimeField";
export { default as DateTimeField } from "./DateTimeField";
export { default as ListField } from "./ListField";
export { default as DataField } from "./DataField";
export { default as TabsField } from "./TabsField";
export { default as SubmitIconField } from "./SubmitIconField";
export { default as SubmitFabField } from "./SubmitFabField";
export { default as FabField } from "./FabField";
export { default as ImageField } from "./ImageField";
export { default as ButtonGroupField } from "./ButtonGroupField";
