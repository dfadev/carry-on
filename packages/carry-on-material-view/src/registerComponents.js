import React from "react";
import { register } from "carry-on-store";
import { Box, Paper, Grid } from "@mui/material";
import TextField from "./TextField";
import CheckboxField from "./CheckboxField";
import ButtonField from "./ButtonField";
import IconButtonField from "./IconButtonField";
import CurrencyField from "./CurrencyField";
import SSNField from "./SSNField";
import EINField from "./EINField";
import PercentageField from "./PercentageField";
import PhoneField from "./PhoneField";
import ZipCodeField from "./ZipCodeField";
import SubmitButtonField from "./SubmitButtonField";
import ResetButtonField from "./ResetButtonField";
import SelectField from "./SelectField";
import RadioField from "./RadioField";
import SwitchField from "./SwitchField";
import ToggleButtonGroupField from "./ToggleButtonGroupField";
import ContentField from "./ContentField";
import InspectorField from "./InspectorField";
import DateField from "./DateField";
import DobField from "./DobField";
import TimeField from "./TimeField";
import DateTimeField from "./DateTimeField";
import ListField from "./ListField";
import DataField from "./DataField";
import TabsField from "./TabsField";
import SubmitIconField from "./SubmitIconField";
import SubmitFabField from "./SubmitFabField";
import FabField from "./FabField";
import ImageField from "./ImageField";
import ButtonGroupField from "./ButtonGroupField";

function FormViewBox({ children }) {
  return (
    <Paper elevation={3}>
      <Box m={1} p={1}>
        {children}
      </Box>
    </Paper>
  );
}

function SectionBox({ section, children, ...props }) {
  return <Box {...props}>{children}</Box>;
}

function View({ children, ...props }) {
  return (
    <Grid {...props} container>
      {children}
    </Grid>
  );
}

function ViewItem({ children, field, ...props }) {
  return (
    <Grid {...props} item>
      <Box mb={1} p={1}>
        {children}
      </Box>
    </Grid>
  );
}

export const materialViewComponents = {
  components: {
    FormViewBox,
    SectionBox,
    View,
    ViewItem,
    editors: {
      button: ButtonField,
      buttonGroup: ButtonGroupField,
      checkbox: CheckboxField,
      content: ContentField,
      currency: CurrencyField,
      data: DataField,
      date: DateField,
      dateTime: DateTimeField,
      dob: DobField,
      ein: EINField,
      fab: FabField,
      iconButton: IconButtonField,
      image: ImageField,
      inspector: InspectorField,
      list: ListField,
      percentage: PercentageField,
      phone: PhoneField,
      radio: RadioField,
      reset: ResetButtonField,
      select: SelectField,
      ssn: SSNField,
      submit: SubmitButtonField,
      submitFab: SubmitFabField,
      submitIcon: SubmitIconField,
      switch: SwitchField,
      tabs: TabsField,
      text: TextField,
      time: TimeField,
      toggleButtonGroup: ToggleButtonGroupField,
      zipCode: ZipCodeField
    }
  }
};

function registerComponents(...args) {
  register({ state: materialViewComponents }, ...args);
}

export default registerComponents;
