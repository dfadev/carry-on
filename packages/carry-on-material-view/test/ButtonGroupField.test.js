/**
 * @jest-environment jsdom
 */
import React from "react";
import { initStores } from "carry-on-store";
import { render } from "@testing-library/react";
import { Paper, Box } from "@material-ui/core";
import { Store, Register } from "carry-on-react";
import { Form } from "carry-on-react-forms";
import {
  FormView,
  Fields,
  Field,
  Sections,
  Section
} from "carry-on-react-view";
import { materialViewComponents } from "../src/registerComponents";
import ButtonGroupField from "../src/ButtonGroupField";

beforeEach(() => {
  initStores();
});

it("should render <ButtonGroupField> inside a Form", () => {
  let clickCount = 0;
  expect(
    render(
      <Store id="buttonGroupForm">
        <Form>
          <Register>
            {() => ({
              clickHandler: (e, store) => {
                clickCount += 1;
              }
            })}
          </Register>
          <Paper>
            <Box px={3} pt={2} pb={3} mb={2}>
              <ButtonGroupField
                name="abc"
                buttons={[
                  {
                    value: 1,
                    content: <Box style={{ color: "red" }}>Red</Box>,
                    onClick: "clickHandler"
                  },
                  {
                    value: 2,
                    content: <Box style={{ color: "green" }}>Green</Box>,
                    onClick: "clickHandler"
                  },
                  {
                    value: 3,
                    content: <Box style={{ color: "blue" }}>Blue</Box>,
                    onClick: "clickHandler"
                  }
                ]}
              />
            </Box>
          </Paper>
        </Form>
      </Store>
    ).asFragment()
  ).toMatchSnapshot();
});

it("should render <ButtonGroupField> inside a FormView", () => {
  let clickCount = 0;
  expect(
    render(
      <Store id="buttonGroupFormView">
        <FormView>
          <Register>{materialViewComponents}</Register>
          <Fields>
            <Field
              name="abc"
              editor="buttonGroup"
              buttons={[
                { value: 1, content: <Box style={{ color: "red" }}>Red</Box> },
                {
                  value: 2,
                  content: <Box style={{ color: "green" }}>Green</Box>
                },
                { value: 3, content: <Box style={{ color: "blue" }}>Blue</Box> }
              ]}
            />
          </Fields>
          <Sections>
            <Section>{["abc"]}</Section>
          </Sections>
        </FormView>
      </Store>
    ).asFragment()
  ).toMatchSnapshot();
});
