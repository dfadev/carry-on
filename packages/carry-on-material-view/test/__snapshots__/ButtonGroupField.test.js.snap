// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <ButtonGroupField> inside a Form 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.emotion-1 {
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 16px;
  padding-bottom: 24px;
  margin-bottom: 16px;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
}

.emotion-2 .MuiButtonGroup-grouped {
  min-width: 40px;
}

.emotion-2 .MuiButtonGroup-grouped:not(:first-of-type) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  margin-left: -1px;
}

.emotion-2 .MuiButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right-color: transparent;
}

.emotion-2 .MuiButtonGroup-grouped:not(:last-of-type):hover {
  border-right-color: currentColor;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(25, 118, 210, 0.5);
  color: #1976d2;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
  border: 1px solid #1976d2;
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<form
    id="buttonGroupForm.form"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-0"
    >
      <div
        class="MuiBox-root emotion-1"
      >
        <div
          class="MuiButtonGroup-root MuiButtonGroup-outlined emotion-2"
          role="group"
          value=""
        >
          <button
            class="MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary emotion-3"
            id="buttonGroupForm.form.abc[0]"
            tabindex="0"
            type="button"
            value="1"
          >
            <div
              class="MuiBox-root emotion-4"
              style="color: red;"
            >
              Red
            </div>
            <span
              class="MuiTouchRipple-root emotion-5"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary emotion-3"
            id="buttonGroupForm.form.abc[1]"
            tabindex="0"
            type="button"
            value="2"
          >
            <div
              class="MuiBox-root emotion-4"
              style="color: green;"
            >
              Green
            </div>
            <span
              class="MuiTouchRipple-root emotion-5"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary emotion-3"
            id="buttonGroupForm.form.abc[2]"
            tabindex="0"
            type="button"
            value="3"
          >
            <div
              class="MuiBox-root emotion-4"
              style="color: blue;"
            >
              Blue
            </div>
            <span
              class="MuiTouchRipple-root emotion-5"
            />
          </button>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`should render <ButtonGroupField> inside a FormView 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);
}

.emotion-1 {
  margin: 8px;
  padding: 8px;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-5 {
  margin-bottom: 8px;
  padding: 8px;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
}

.emotion-6 .MuiButtonGroup-grouped {
  min-width: 40px;
}

.emotion-6 .MuiButtonGroup-grouped:not(:first-of-type) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  margin-left: -1px;
}

.emotion-6 .MuiButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right-color: transparent;
}

.emotion-6 .MuiButtonGroup-grouped:not(:last-of-type):hover {
  border-right-color: currentColor;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(25, 118, 210, 0.5);
  color: #1976d2;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
  border: 1px solid #1976d2;
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: transparent;
  }
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-9 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<form
    id="buttonGroupFormView.form"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 emotion-0"
    >
      <div
        class="MuiBox-root emotion-1"
      >
        <div
          class="MuiBox-root emotion-2"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item emotion-4"
            >
              <div
                class="MuiBox-root emotion-5"
              >
                <div
                  class="MuiButtonGroup-root MuiButtonGroup-outlined emotion-6"
                  role="group"
                  value=""
                >
                  <button
                    class="MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary emotion-7"
                    id="buttonGroupFormView.form.abc[0]"
                    tabindex="0"
                    type="button"
                    value="1"
                  >
                    <div
                      class="MuiBox-root emotion-2"
                      style="color: red;"
                    >
                      Red
                    </div>
                    <span
                      class="MuiTouchRipple-root emotion-9"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary emotion-7"
                    id="buttonGroupFormView.form.abc[1]"
                    tabindex="0"
                    type="button"
                    value="2"
                  >
                    <div
                      class="MuiBox-root emotion-2"
                      style="color: green;"
                    >
                      Green
                    </div>
                    <span
                      class="MuiTouchRipple-root emotion-9"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary emotion-7"
                    id="buttonGroupFormView.form.abc[2]"
                    tabindex="0"
                    type="button"
                    value="3"
                  >
                    <div
                      class="MuiBox-root emotion-2"
                      style="color: blue;"
                    >
                      Blue
                    </div>
                    <span
                      class="MuiTouchRipple-root emotion-9"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
